src/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -419,7 +419,9 @@ protected function processParams(File $phpcsFile, $stackPtr, $commentStart)
 
                 // Check type hint for array and custom type.
                 $suggestedTypeHint = '';
-                if (strpos($suggestedName, 'array') !== false || substr($suggestedName, -2) === '[]') {
+                if (preg_match('/^(.*)<.*>$/', $suggestedName, $matches)) {
```

Comment:
1. For performance optimization, I'd recommend having this condition as the _last_ condition in the if/elseif chain, not the first.
2. I'd also suggest optimizing the regex a little more, like so:
    ```suggestion
                    if (preg_match('/^([^<]+)<[^>]+>$/', $suggestedName, $matches)) {
    ```